api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.config.common.mutation_rules.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.config.common.mutation_rules.v3";
+option java_outer_classname = "MutationRulesProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// The HeaderMutationRules structure specifies what headers may be
+// manipulated by a processing filter. By default, all mutations are
+// possible. This rule makes it possible to disallow modifications of
+// certain filters according to the parameters below.
+//
+// This structure is used by the External Processing Filter to make
+// it possible to restrict what an external processing server can do.
+// Every attempt to add, change, append, or remove a header will be
+// tested against the rules here. Disallowed header mutations will be
+// ignored, and a DEBUG message logged, unless "disallow_is_error" is
+// set to true.
+// [#next-free-field: 8]
+message HeaderMutationRules {
+  // Prevent modification of headers that could affect routing
+  // decisions. This includes "host", ":authority", ":scheme",
+  // and ":method".
+  bool disallow_routing = 1;
+
+  // Prevent modification of envoy-internal headers. These typically
+  // start with "x-envoy" but this prefix may be overridden elsewhere.
+  bool disallow_envoy = 2;
+
+  // Prevent modification of any system header, defined as a header that
+  // starts with a ":" character.
+  bool disallow_system = 3;
+
+  // Prevent modification of any header at all. For the ext_proc filter,
+  // this means that a processing server may still choose to reject the
+  // entire request.
+  bool disallow_all = 4;
+
+  // If set, specifically allow any header that matches this regular
+  // expression. This overrides all other settings except for
+  // "disallow_expression".
+  string allow_expression = 5;
```

Comment:
Consider use of RegexMatcher.  These regexs should be RE2

https://github.com/envoyproxy/envoy/blob/main/api/envoy/type/matcher/regex.proto#L18

Reply:
Yes that makes a lot of sense -- I pushed an update. Thanks!